### PR TITLE
[IMP] mail: add the "raise hand" feature to calls

### PR DESCRIPTION
**Pull request by ThanhDodeurOdoo**
This commit adds a button to the `CallActionList` to signal to the other participants of the call that one is raising a hand.

This will add a hand icon to the `CallParticipantCard` of the people with a raised hand and send a notification that is displayed in the `Call` view (only when in discuss, not in chat windows).

task-3229507

### CODE REVIEW

File: addons/mail/static/src/rtc/rtc_service.js [hunk 1]
```diff
@@ -403,6 +404,19 @@ export class Rtc {
...
                 this.disconnect(session);
                 break;
+            case "raisingHandChange":
+                Object.assign(session, { isRaisingHand: payload.isRaisingHand });
+                // eslint-disable-next-line no-case-declarations
+                const notificationId = "raise_hand_" + session.id;
+                if (session.isRaisingHand) {
+                    this.callNotification.add({
+                        id: notificationId,
+                        text: sprintf("%s raised a hand", session.name),
```
Comment (alexkuhn) [inline addons/mail/static/src/rtc/rtc_service.js:414]: Translation

File: addons/mail/static/src/rtc/rtc_service.js [hunk 2]
```diff
@@ -641,6 +663,9 @@ export class Rtc {
                         isSelfMuted: this.state.selfSession.isSelfMuted,
                     },
                 });
+                await this.notify([session], "raisingHandChange", {
```
Comment (alexkuhn) [inline addons/mail/static/src/rtc/rtc_service.js:666]: I prefer `raise_hand` for notif name.

File: addons/mail/static/src/rtc/rtc_service.js [hunk 3]
```diff
@@ -641,6 +663,9 @@ export class Rtc {
...
+                    isRaisingHand: this.state.selfSession.isRaisingHand,
```
Comment (alexkuhn) [inline addons/mail/static/src/rtc/rtc_service.js:667]: Just `active` for named parameter

File: addons/mail/static/src/rtc/rtc_service.js [hunk 4]
```diff
@@ -435,6 +449,14 @@ export class Rtc {
         this.soundEffectsService.play("mute");
     }
 
+    async raiseHand() {
+        await this.setIsRaisingHand(true);
+    }
+
+    async lowerHand() {
+        await this.setIsRaisingHand(false);
+    }
```
Comment (alexkuhn) [inline addons/mail/static/src/rtc/rtc_service.js:458]: Not need to split in 2 functions.
Can just be a single function `raiseHand(true/false)`.
This solves issue of making a weirdly named function `setIsRaisingHand`.

File: addons/mail/static/src/rtc/call_participant_card.xml [hunk 5]
```diff
@@ -36,6 +36,9 @@
                     </small>
                 </span>
                 <div class="o-mail-CallParticipantCard-overlay position-absolute top-0 end-0 d-flex flex-row-reverse">
+                    <span t-if="rtcSession.isRaisingHand" class="d-flex flex-column justify-content-center me-1 rounded-circle bg-500" t-att-class="{'o-minimized p-1': props.minimized, 'p-2': !props.minimized }" title="raising hand" aria-label="raising hand">
```
Comment (alexkuhn) [inline addons/mail/static/src/rtc/call_participant_card.xml:39]: `o-minimized` looks unused (some other occurrences in this template are also unused it seems)

Comment (ThanhDodeurOdoo) [reply addons/mail/static/src/rtc/call_participant_card.xml:39]: @alexkuhn it used to have the `p-1` equivalent in scss and I think that the design team kept the class for backwards compatibility when they refactored

Comment (alexkuhn) [reply addons/mail/static/src/rtc/call_participant_card.xml:39]: Then can remove it, if unused then not worth keeping it ;)

**Comment by alexkuhn** [comment]
Don't forget to register the new service in `mail/tests/webclient_setup.js`

Screenshot 2023-03-27 at 12 12 26

**Pull request closed without merging**